Mathlib/Topology/Irreducible.lean

```diff
@@ -311,4 +313,80 @@ theorem IsPreirreducible.preimage (ht : IsPreirreducible t) {f : Y → X}
   cases hf.injective h₄
   exact ⟨y, h₁, h₂, h₃⟩
 
+section
+
+open Set.Notation
+
+@[stacks 004Z]
+lemma IsPreirreducible.preimage_of_dense_isPreirreducible_fiber
+    {V : Set Y} (hV : IsPreirreducible V)
+    (f : X → Y) (hf' : IsOpenMap f) (hf'' : Dense (V ↓∩ { x | IsPreirreducible (f ⁻¹' {x}) })) :
+    IsPreirreducible (f ⁻¹' V) := by
```

Comment:
Wouldn't this be easier to use if `hf''` was stated as `V` being inside the closure of `V ∩ { x | IsPreirreducible (f ⁻¹' {x}` ? This avoids bringing subtypes into the party, but I don't know enough about the applications to be certain that it's better.